Wikipedia/Code/WMFAlertManager.swift

```diff
@@ -94,12 +94,29 @@ open class WMFAlertManager: NSObject, RMessageProtocol, Themeable {
         })
     }
 
-    func showBottomAlertWithMessage(_ message: String, subtitle: String?, image: UIImage?, type: RMessageType, customTypeName: String?, duration: TimeInterval? = nil, dismissPreviousAlerts:Bool, callback: (() -> Void)? = nil, buttonTitle: String? = nil, buttonCallBack: (() -> Void)? = nil) {
-        showAlert(dismissPreviousAlerts, alertBlock: { () in
-            RMessage.showNotification(withTitle: message, subtitle: subtitle, iconImage: image, type: type, customTypeName: customTypeName, duration: duration ?? 5, callback: callback, buttonTitle: buttonTitle, buttonCallback: buttonCallBack, at: .bottom, canBeDismissedByUser: true)
-        })
+    func showBottomAlertWithMessage(_ message: String, subtitle: String?, image: UIImage?, type: RMessageType, customTypeName: String?, duration: TimeInterval? = nil, dismissPreviousAlerts: Bool, callback: (() -> Void)? = nil, buttonTitle: String? = nil, buttonCallBack: (() -> Void)? = nil, completion: (() -> Void)? = nil
+    ) {
+        showAlert(dismissPreviousAlerts) {
+            RMessage.showNotification(
+                withTitle: message,
+                subtitle: subtitle,
+                iconImage: image,
+                type: type,
+                customTypeName: customTypeName,
+                duration: duration ?? 5,
+                callback: callback,
+                buttonTitle: buttonTitle,
+                buttonCallback: buttonCallBack,
+                at: .bottom,
+                canBeDismissedByUser: true
+            )
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + (duration ?? 5)) {
```

Comment:
One possible improvement you could try at the call site is set `dismissPreviousAlerts: false` on the second toast instead of chaining them and using this `asyncAfter`. I'm not sure how well it works though, so if this doesn't work I'm fine with what you did here.

```
// Present edit published
WMFAlertManager.sharedInstance.showBottomAlertWithMessage(...dismissPreviousAlerts: true...)

// Present temp account
 if let needsNewTempAccountToast,  needsNewTempAccountToast {
   WMFAlertManager.sharedInstance.showBottomAlertWithMessage(...dismissPreviousAlerts: false...)
}
```